api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java

```diff
@@ -95,4 +104,40 @@
     log.info(String.format("successfully audited %d users", request.getUserIds().size()));
     return ResponseEntity.noContent().build();
   }
+
+  @Override
+  public ResponseEntity<Void> synchronizeUserAccess(SynchronizeUserAccessRequest request) {
+    int errorCount = 0;
+    for (long userId : request.getUserIds()) {
+      DbUser user = userDao.findUserByUserId(userId);
+
+      try {
+        // 2FA synchronization requires an outgoing call to gsuite. For this reason, we
+        // optimize to only verify that users who have 2FA enabled, still have it enabled.
+        // Users who don't have 2FA will go through an active flow to enable it, and are not
+        // dependent on this offline check. Disabled users have no access anyways, so don't
+        // bother checking them either.
+        if (user.getTwoFactorAuthCompletionTime() != null && !user.getDisabled()) {
+          user = userService.syncTwoFactorAuthStatus(user, Agent.asSystem());
+        }
+
+        // Always synchronize for consistency. Under normal operation only the passage of time
+        // should result in access expiration changes here, but this is also a backstop for ensuring
+        // the database is consistent with our access tier groups, e.g. due to partial system
+        // failures or bugs.
+        userService.updateUserWithRetries(Function.identity(), user, Agent.asSystem());
+      } catch (WorkbenchException e) {
```

Comment:
Yes. Any other expected exceptions should be checked, which aren't getting to us here. The next level I could catch beyond this would be `Exception`, which is a bad pattern as it can obscure programming errors like NPE.

In an ideal world, I think these internal services would instead throw an internal checked excpetion, instead of the unchecked `WorkbenchException`, but this is what I have for now.